packages/js/src/generators/library/library.ts

```diff
@@ -315,11 +337,14 @@ function createFiles(tree: Tree, options: NormalizedSchema, filesDir: string) {
       };
       return json;
     });
-  } else if (!options.bundler || options.bundler === 'none') {
+  } else if (
+    (!options.bundler || options.bundler === 'none') &&
+    !(options.projectRoot === '.')
+  ) {
     tree.delete(packageJsonPath);
   }
 
-  if (options.minimal) {
+  if (options.minimal && !(options.projectRoot === '.')) {
```

Comment:
Should check `options.rootProject` instead of the root? Also, would this delete the `README.md` generated for the workspace?

Maybe the `README.md` file should pulled from `files/lib` and generated separately, like:

```ts
if (!options.minimal && !options.rootProject) {
  generateFiles(tree, join(__dirname, 'files/with-readme', {...});
}
```

Reply:
Jason wants to minimize the use of `options.rootProject`, and suggested checking for the project root instead. This check explicitly prevents deleting the the readme if passed minimal and it is the root project.